src/omezarr-helper.ts

```diff
@@ -500,141 +507,106 @@ async function getOmeZarrThumbnail(
-    const startIndices: number[] = [];
-    const endIndices: number[] = [];
-
-    for (let i = 0; i < arrayShape.length; i++) {
-      const chunkSize = chunks[i];
-      log.trace('Chunk size:', chunkSize);
-      const center = centerPoint[i];
-      log.trace('Center:', center);
-
-      // Find which chunk contains the center point
-      const centerChunkIndex = Math.floor(center / chunkSize);
-      log.trace('Center chunk index:', centerChunkIndex);
-
-      // Calculate the start and end indices
-      startIndices[i] = centerChunkIndex * chunkSize;
-      log.trace('Start index:', startIndices[i]);
-      endIndices[i] = Math.min(startIndices[i] + cropSize, arrayShape[i]);
-      log.trace('End index:', endIndices[i]);
-    }
-
-    // Create selection slice for the crop
-    const selection = startIndices.map((start, i) => [start, endIndices[i]]);
-
-    log.debug(
-      'Crop dimensions:',
-      selection.map(([start, end]) => end - start)
-    );
+  return new Promise((resolve, reject) => {
+    const img = new Image();
+    img.onload = () => {
+      try {
+        const canvas = document.createElement('canvas');
+        const ctx = canvas.getContext('2d');
+        if (!ctx) {
+          reject(new Error('Could not get canvas context'));
+          return;
+        }
 
-    // Fetch the crop data using zarrita's get API
-    const cropSelection = selection.map(([start, end]) =>
-      zarr.slice(start, end)
-    );
-    const cropData = await zarr.get(arr, cropSelection);
-
-    // Convert to typed array for easier processing
-    let flatData: ArrayLike<number>;
-    if (
-      cropData.data instanceof ArrayBuffer ||
-      ArrayBuffer.isView(cropData.data)
-    ) {
-      flatData = new Float32Array(cropData.data as ArrayBuffer);
-    } else if (Array.isArray(cropData.data)) {
-      flatData = cropData.data as number[];
-    } else {
-      // Handle TypedArray case
-      flatData = cropData.data as ArrayLike<number>;
-    }
-    const totalValues = flatData.length;
+        // Get original image data
+        canvas.width = img.width;
+        canvas.height = img.height;
+        ctx.drawImage(img, 0, 0);
+        const origData = ctx.getImageData(0, 0, img.width, img.height);
+
+        // Clear canvas and draw shifted image (1 pixel to the right)
+        ctx.clearRect(0, 0, canvas.width, canvas.height);
+        ctx.drawImage(img, 1, 0);
+        const shiftData = ctx.getImageData(0, 0, img.width, img.height);
+
+        let nonZeroPixels = 0;
+        const totalPixels = img.width * img.height;
+
+        // Compare original and shifted images pixel by pixel
+        for (let i = 0; i < origData.data.length; i += 4) {
+          // Calculate difference for RGB channels (ignore alpha)
+          const rDiff = Math.abs(origData.data[i] - shiftData.data[i]);
+          const gDiff = Math.abs(origData.data[i + 1] - shiftData.data[i + 1]);
+          const bDiff = Math.abs(origData.data[i + 2] - shiftData.data[i + 2]);
```

Comment:
That's probably not a concern for an image a few hundred pixels in size, but just in case, I think, given how the shifting is done `ctx.drawImage(img, 1, 0)`, there will always be an artificial edge at x == 0 because the subtraction is done over the entire canvas, without excluding the x == 0 column?

Reply:
Right, and the extra edge probably changes the ratio by something like `1 / width`, so around 0.01 or 0.001, depending on the image. And the cutoff is 0.05 so it should be fine..